src/gui/render/array_buffer/array_buffer.hpp

```diff
@@ -41,143 +41,153 @@ class ArrayBuffer;
+                assert(false && "Invalid type.");
+                break;
         }
     }
 
     template <typename T> // float/ double
-    constexpr inline static GPUDataType presume_type() {
+    constexpr inline static GPUDataType
+    presume_type() {
         switch (sizeof(T)) {
-        case sizeof(glm::float32):
-        
+            case sizeof(glm::float32):
+
                 return GPUDataType::FLOAT;
 
-        case sizeof(glm::float64):
+            case sizeof(glm::float64):
                 return GPUDataType::DOUBLE;
 
-        default:
-            assert(false && "Invalid type.");
-            break;
+            default:
+                assert(false && "Invalid type.");
+                break;
         }
     }
 
     template <std::integral T>
-    inline constexpr static array_buffer_type init([[maybe_unused]] T i) {
+    inline constexpr static array_buffer_type
+    init([[maybe_unused]] T i) {
         return array_buffer_type(1, sizeof(T), true, presume_type<T>());
     }
 
     template <std::floating_point T>
-    inline constexpr static array_buffer_type init([[maybe_unused]] T i) {
+    inline constexpr static array_buffer_type
+    init([[maybe_unused]] T i) {
         return array_buffer_type(1, sizeof(T), false, presume_type<T>());
     }
 
     template <int i, class T, glm::qualifier Q>
-    constexpr static array_buffer_type init([[maybe_unused]] glm::vec<i, T, Q> V){
-        return array_buffer_type(i,
-            sizeof(T),
-            std::is_integral_v<T>,
-            presume_type<T>());
-        }
+    constexpr static array_buffer_type
+    init([[maybe_unused]] glm::vec<i, T, Q> V) {
+        return array_buffer_type(
+            i, sizeof(T), std::is_integral_v<T>, presume_type<T>()
+        );
+    }
 
     template <class T>
-    constexpr static array_buffer_type init([[maybe_unused]] T t){
+    constexpr static array_buffer_type
+    init([[maybe_unused]] T t) {
         assert(false && "Invalid type.");
-        return array_buffer_type(0,0,0,GPUDataType::BYTE);
+        return array_buffer_type(0, 0, 0, GPUDataType::BYTE);
     }
 
     bool operator==(const array_buffer_type& other) const = default;
 
     bool operator==(const ArrayBuffer& other) const;
-
 };
 
 class ArrayBuffer : public array_buffer_type {
  private:
     GLuint buffer_ID_;
 
-    GLuint devisor_; // for instancing
+    GLuint divisor_; // for instancing
 
     const buffer_type buffer_type_;
 
-
  public:
-
     template <class T>
-    void update(std::vector<T> data, GLuint devisor);
+    void update(std::vector<T> data, GLuint divisor);
 
     template <class T>
-    inline void update(std::vector<T> data ) {update<T>(data, devisor_ ); }
+    inline void
+    update(std::vector<T> data) {
+        update<T>(data, divisor_);
+    }
 
-    inline void set_devisor(GLuint devisor) noexcept {devisor_ = devisor;}
+    inline void
+    set_divisor(GLuint divisor) noexcept {
+        divisor_ = divisor;
+    }
```

Comment:
If you're not doing any checks just do this
```suggestion
    inline auto&
    divisor(GLuint divisor) noexcept {
        return divisor_;
    }
```